tests/e2e/support/objects/app-files/link/actions.ts

```diff
@@ -72,23 +78,30 @@ const deleteLinkButton =
 const confirmDeleteButton = `//button[contains(@class,"oc-modal-body-actions-confirm") and text()="Delete"]`
 
 export const createLink = async (args: createLinkArgs): Promise<string> => {
-  const { page, resource } = args
-  const resourcePaths = resource.split('/')
-  const resourceName = resourcePaths.pop()
-  if (resourcePaths.length) {
-    await clickResource({ page: page, path: resourcePaths.join('/') })
+  const { space, page, resource } = args
+  if (!space) {
+    const resourcePaths = resource.split('/')
+    const resourceName = resourcePaths.pop()
+    if (resourcePaths.length) {
+      await clickResource({ page: page, path: resourcePaths.join('/') })
+    }
+    await sidebar.open({ page: page, resource: resourceName })
+    await sidebar.openPanel({ page: page, name: 'sharing' })
   }
-
-  await sidebar.open({ page: page, resource: resourceName })
-  await sidebar.openPanel({ page: page, name: 'sharing' })
   await page.locator(addPublicLinkButton).click()
+  await waitForPopupNotPresent(page)
   // const message = await page.locator(linkUpdateDialog).textContent()
   // expect(message.trim()).toBe('Link was created successfully')
   // const linkId = await page.getAttribute(getMostRecentLink, 'data-testid')
   // return linkId.replace('files-link-id-', '')
   return await page.locator(util.format(publicLink, 'Link')).textContent()
 }
 
+export const waitForPopupNotPresent = async (page): Promise<void> => {
```

Comment:
is used only once here, can you please inline it

Reply:
The function could have been inlined, but I thought it would be better to create a separate function so that it can be reused in the future if needed.

Reply:
Thanks, fine for me to have it as it is for later reuse :)